### DISCUSSION TITLE
Add prop onChange to all date \ time pickers

### DISCUSSION THREAD

**Discussion by OliverWall** [lookfirst/mui-rff#877] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I need to store the time in YYYY-DD-MM format in FinalForm. 

**Describe the solution you'd like**
Allow passing props onChange as callback for transformation before changing the form state

**Describe alternatives you've considered**
I tried to use transform in Yup lib but it doesn't fit my needs.

**Additional context**
No addithional context

**Comment by lookfirst** [comment, marked as answer]
Hi @OliverWall, I moved this to a discussion since it isn't an 'issue' as of yet.

Can you please explain a bit of your question first though? I'm not really sure what you mean. Why does it need to be stored in FinalForm as a specific format?

**Reply by OliverWall** [reply]
Hi @lookfirst.

Thank you so much for the answer and your lib.

I needed to store and send the date to the server in different formats. With and without timezones. As a result, I managed to solve this problem through several transformations using Yup.
I have finalized the scheme for initialization and transformation to send data to the server. I just didn't have enough knowledge on Yup to solve this problem.

**Comment by lookfirst** [comment]
I'm glad you figured it out @OliverWall! Thanks for the update.
